samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -274,7 +276,8 @@ public void onContainerFailed(Throwable t) {
               } else {
                 LOGGER.warn("JobCoordinatorLatch was null. It is possible for some component to be waiting.");
               }
-              LOGGER.error("Container failed. Stopping the processor.", t);
+              containerException = t;
```

Comment:
All this logic becomes very confusing with a different ways of invoking stop() from different points of the code.
May be we should think about some kind of doStop(Throwable t) - to differentiate between different stops, instead trying to figure every time why are we stopping - for an error, or normal.